### PR TITLE
XW-1249 | send document title for category pages on Mercury

### PR DESCRIPTION
**Pull request by tomasznapieralski**
https://wikia-inc.atlassian.net/browse/XW-1249

/cc @Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 1]
```diff
@@ -363,13 +363,14 @@ public function getPage() {
...
 							$data['details'] = MercuryApiArticleHandler::getArticleDetails( $article );
-							$titleBuilder->setParts( [$data['article']['displayTitle']] );
 						} elseif ( !empty( $data['nsSpecificContent']['members']['sections'] ) ) {
 							$data['details'] = MercuryApiCategoryHandler::getCategoryMockedDetails( $title );
-							$titleBuilder->setParts( [$title->getPrefixedText()] );
 						} else {
 							throw new NotFoundApiException( 'Article is empty and category has no members' );
 						}
+
+						$data['details']['documentTitle'] = htmlspecialchars( $article->getTitle()->getPrefixedText() );
```
Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: You can use `$title` here.

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 2]
```diff
@@ -331,6 +331,7 @@ public function getArticleFromMarkup() {
 	public function getPage() {
 		try {
 			$title = $this->getTitleFromRequest();
+			$documentTitle = '';
```
Comment (bognix) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: remove this set

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 3]
```diff
@@ -354,6 +354,8 @@ public function getPage() {
 				} else {
 					$data['details'] = MercuryApiMainPageHandler::getMainPageMockedDetails( $title );
 				}
+
+				$documentTitle = '';
```
Comment (bognix) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: add comment why you set it to empty string for main pages

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 4]
```diff
@@ -354,6 +353,9 @@ public function getPage() {
 				} else {
 					$data['details'] = MercuryApiMainPageHandler::getMainPageMockedDetails( $title );
 				}
+
+				// On Curated Main Pages page title and Wiki name are equal and to avoid duplicates we don't use page title
+				$documentTitle = '';
```
Comment (bognix) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: on not curated main pages it's still going to be broken

Comment (tomasznapieralski) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: good catch

Comment (tomasznapieralski) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: fixed

Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: After https://github.com/Wikia/app/pull/10110/files#diff-551276c093245208cefecd540a00ba55R334 it's already an empty string.

**Comment by rogatty** [comment]
Fix small duplication and will LGTM

**Comment by bognix** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

**Pull request merged**